Source/JavaScriptCore/API/JSValueRef.h

```diff
@@ -292,6 +320,140 @@ JS_EXPORT JSValueRef JSValueMakeString(JSContextRef ctx, JSStringRef string);
+
+/*!
+    @function
+    @abstract         Creates a JavaScript BigInt with a 64-bit signed integer.
+    @param ctx        The execution context to use.
+    @param integer    The 64-bit signed integer to copy into the new BigInt JSValue.
+    @param exception  A pointer to a JSValueRef in which to store an exception, if any. Pass NULL if you do not care to store an exception.
+    @result           A BigInt JSValue of the integer, or NULL if an exception is thrown.
+*/
+JS_EXPORT JSValueRef JSBigIntCreateWithInt64(JSContextRef ctx, int64_t integer, JSValueRef* exception) JSC_API_AVAILABLE(macos(JSC_MAC_TBA), ios(JSC_IOS_TBA));
+
+/*!
+    @function
+    @abstract         Creates a JavaScript BigInt with an integer represented in string.
+    @param ctx        The execution context to use.
+    @param string     The JSStringRef representation of an integer.
+    @param exception  A pointer to a JSValueRef in which to store an exception, if any. Pass NULL if you do not care to store an exception.
+    @result           A BigInt JSValue of the string, or NULL if an exception is thrown.
+    @discussion       This is equivalent to calling the `BigInt` constructor from JavaScript with a string argument.
+*/
+JS_EXPORT JSValueRef JSBigIntCreateWithString(JSContextRef ctx, JSStringRef string, JSValueRef* exception) JSC_API_AVAILABLE(macos(JSC_MAC_TBA), ios(JSC_IOS_TBA));
+
+/*!
+    @function
+    @abstract         Converts a JSValue to an unsigned 64-bit integer and returns the resulting integer.
+    @param ctx        The execution context to use.
+    @param value      The JSValue to convert.
+    @param exception  A pointer to a JSValueRef in which to store an exception, if any. Pass NULL if you do not care to store an exception.
+    @result           A uint64_t with the result of conversion, or 0 if an exception is thrown.
+    @discussion       Convert the JSValue to a numeric by `ToNumeric` in JavaScript first. Then apply `ToBigUint64` if the numeric is a BigInt, otherwise do the truncation similar to `ToUint32` but for int64.
+*/
+JS_EXPORT uint64_t JSValueToUInt64(JSContextRef ctx, JSValueRef value, JSValueRef* exception) JSC_API_AVAILABLE(macos(JSC_MAC_TBA), ios(JSC_IOS_TBA));
+
+/*!
+    @function
+    @abstract         Converts a JSValue to a singed 64-bit integer and returns the resulting integer.
+    @param ctx        The execution context to use.
+    @param value      The JSValue to convert.
+    @param exception  A pointer to a JSValueRef in which to store an exception, if any. Pass NULL if you do not care to store an exception.
+    @result           A int64_t with the result of conversion, or 0 if an exception is thrown.
+    @discussion       Convert the JSValue to a numeric by `ToNumeric` in JavaScript first. Then apply `ToBigInt64` if the numeric is a BigInt, otherwise do the truncation similar to `ToInt32` but for int64.
```

Comment:
The JSValue is converted to an integer according to the rules specified by the JavaScript language. If the value is a BigInt then the value is truncated to an `int64_t`